src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1002,5 +1158,75 @@ private void AddSample<T>(List<T> samples, T sample)
                 samples.Add(sample);
             }
         }
+
+        private sealed class PendingItem
```

Comment:
Is this essentially a pending grpc message / event we want to handle? I think a more descriptive name could be helpful. If we change the name here it might also be good to make sure other methods such as `ProcessItem` match i.e. ProcessGrpcEvent

Also wondering if we should move this into it's own file